app/models/user.rb

```diff
@@ -349,6 +345,17 @@ def clear_authentication_token!
     save!
   end
 
+  def expired_authentication_token?
+    return true if authentication_token.blank?
+
+    last_activity = [current_sign_in_at, last_sign_in_at, last_seen_at].compact.max
+
+    return false if last_activity.blank?
```

Comment:
` return false if last_activity.blank?`
what's does this do here? when is last activity going to be blank?

Reply:
the question is: is the auth token expired?

the answer is no (`false`) if the user has never been seen or signed in.

this condition happens for new users. They have a token, but none of the tracker entries have been updated or saved to the database yet.